### PR TITLE
Userprofile

### PR DESCRIPTION
**Pull request by regilkent123**

### CODE REVIEW

**Comment by regilkent123** [comment]
naapil ang .pyc kuya pero ang .sqlite3 kay wala

File: userprofile/migrations/0003_auto_20180521_0055.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 2.0.4 on 2018-05-21 00:55
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
```
Comment (ozknightwalker) [inline userprofile/migrations/0003_auto_20180521_0055.py]: please merge your migrations to 1

File: userprofile/models.py [hunk 2]
```diff
@@ -10,11 +10,14 @@ class UserProfile(models.Model):
             User, on_delete=models.CASCADE,
             primary_key=True,
         )
+
+    trainee = 1
```
Comment (ozknightwalker) [inline userprofile/models.py]: capitalize static variables please

File: userprofile/models.py [hunk 3]
```diff
@@ -10,11 +10,14 @@ class UserProfile(models.Model):
...
+    trainer = 2
     utype = (
-            ('Trainee','Trainee'),
-            ('Trainer','Trainer'),
+            (trainee,'Trainee'),
+            (trainer,'Trainer'),
         )
-    usertype = models.CharField(max_length=7, choices=utype, default='Trainee')
+    usertype = models.IntegerField(choices=utype, default=trainee)
```
Comment (ozknightwalker) [inline userprofile/models.py]: use Positive Integer Field for this one

File: userprofile/templates/userprofile/login.html [hunk 4]
```diff
@@ -2,22 +2,23 @@
 {% extends 'base.html' %}
 {% block content %}
 {% load static %}
-<link rel="stylesheet" type="text/css" href="{% static "userprofile/styles.css" %}" />
+<link rel="stylesheet" type="text/css" href="{% static "userprofile/stylesheets/asd.css" %}" />
```
Comment (ozknightwalker) [inline userprofile/templates/userprofile/login.html]: move this to another block

File: userprofile/views.py [hunk 5]
```diff
@@ -8,48 +8,34 @@
 from django.contrib.auth.forms import UserCreationForm
 from .forms import RegisterForm
 
+# unsay error? wait gi close lagi nimo ang forms
```
Comment (ozknightwalker) [inline userprofile/views.py]: please remove this comments

**Review by ozknightwalker — CHANGES_REQUESTED** [review]

File: userprofile/forms.py [hunk 6]
```diff
@@ -3,44 +3,48 @@
...
-        return password2
+        return self.cleaned_data
 
-    def save(self, commit=True):
+    def save(self):
         user = User.objects.create_user(
-            self.cleaned_data['username'],
-            self.cleaned_data['email'],
-            self.cleaned_data['password1']
+            username = self.cleaned_data.get('username'),
```
Comment (ozknightwalker) [inline userprofile/forms.py]: remove spaces around `=`

File: userprofile/templates/userprofile/loggedin.html [hunk 7]
```diff
@@ -3,7 +3,9 @@
 {% block content %}
 
 <h2>Hi {{full_name}} You are now logged in!</h2>
-<p> Click <a href="logout/"> here </a> to log out.</p>
+<form method="post" action="{% url 'user:login' %}">
+<p> Click <a href="{% url 'user:login' %}"> here </a> to log out.</p>
```
Comment (ozknightwalker) [inline userprofile/templates/userprofile/loggedin.html]: the link should logout user not login @regilkent123 @iamkatrinaanne

File: userprofile/views.py [hunk 8]
```diff
@@ -1,83 +1,64 @@
...
+            username = request.POST['username']
+            password = request.POST['password']
 
-def auth_login(request):
-    if request.method == "POST":
-        username = request.POST['username']
-        password = request.POST['password']
-
-        auth_user = authenticate(username=username, password=password)
+            auth_user = authenticate(username=username, password=password)
```
Comment (ozknightwalker) [inline userprofile/views.py]: @regilkent123 @iamkatrinaanne what is the args for?

**Review by ozknightwalker — CHANGES_REQUESTED** [review]

File: userprofile/views.py [hunk 9]
```diff
@@ -1,83 +1,64 @@
...
-    #
-    #     else:
-    #         form = RegisterForm()
-    #     # only 79 char per line guys hahaha please don't go over 79 characters
-    #     return render(request, '../templates/userprofile/register.html', {'form': form})
-
+            else:
+                form = RegistrationForm()
+            args['form'] = form
+        return render(request, self.template_name, {'form': form}, args)
```
Comment (ozknightwalker) [inline userprofile/views.py]: @regilkent123 @iamkatrinaanne what is the `args` for? you already passed the forms back

**Review by ozknightwalker — CHANGES_REQUESTED** [review]

**Review by ozknightwalker — APPROVED** [review]

**Pull request merged**